### DISCUSSION TITLE
$bvModal equivalent implementation

### DISCUSSION THREAD

**Discussion by parafeu** [cdmoro/bootstrap-vue-3#334] [Ideas]
Related to issue #120 and  #329 

I wanted to start a discussion about the usefulness of implementing a way to control the display of a modal without having direct access to its state.

The most obvious use case is the possibility to control a modal that is not present as a child of the component.

For the current implementation I used the globalProperties, but I could for example make a composable looking like this :

```js
const { show, hide } = useBVModal("id");
```

I don't have a strong opinion on this subject, it's not an essential feature in my opinion

**Comment by aceofwings** [comment]
Saw your pull request. 

I do agree with a composable. And I do not mind global property use either. I encourage light weight usage of components and everyone has different design styles. However if possible global properties may want to also be configurable or chosen to be removed hence the use of a `Plugin / Install`

As long as you use an event bus. I can see you building the above proposed composable.

**Comment by parafeu** [comment]
You mean make a composable for composition  API users and a plugin that adds a global property for option API users?

**Reply by aceofwings** [reply]
Pretty much. You may want to look at Bootstrap vue Modal Plugin to see what it encompasses.

**Comment by xvaara** [comment]
My question was, why would you need to show/hide based on the "id" and not just use the v-model? v-model can be a state that is imported so can be used cross components.

**Reply by xvaara** [reply]
if you need to do it with id, you can build your own composable or mixins (if options api is needed).

**Reply by parafeu** [reply]
That's the purpose of this discuss, I don't need it personally, and I know the alternatives. 
I just want to know if this feature should be implemented. 

@cdmoro's response to issue #120 indicated that he wanted to do so.

**Reply by xvaara** [reply]
Yeah... Sorry if I came off negative. Ever since there is hope for bootstrapVue to build for vue3 with @vue/compat, I think this project should not aim for 100% compatibility. Only where it matters. Otherwise lean into being "a good vue 3 citizen".

any thoughs @cdmoro @mariusa ?

**Comment by mariusa** [comment]
Agreed. Let's strive for Vue 3 best practices for this library, without
looking back.

Marius
